C-Plus-Plus/cp/ratInAMaze.cpp

```diff
@@ -0,0 +1,124 @@
+/*There is a rat who is at the starting point of a matrix and wants to reach
+ to the ending point i.e at the last coordinate of the matrix.
+ 1 denotes the point where it is possible for the mouse to go and
+ 0 denotes the dead end */
+
+#include <iostream>
+#include<bits/stdc++.h>
+using namespace std;
+
+ //checking whether safe to move
+bool issafetomove(int** arr, int x, int y, int row, int col)    
+{
+    if(x<row && y<col && arr[x][y]==1)
+    {
+        return true;
+    }
+    return false;
+}
+
+
+bool ratInMaze(int** arr, int x,int y,int row, int col, int** solution)
+{
+//base condition
+    if(x==row-1 && y==col-1)
+    {
+        solution[x][y]=1;
+        return true;
+    }
+
+    if(issafetomove(arr,x,y,row,col))
+    {
+        solution[x][y]=1;
+
+//mpving in right direction
```

Comment:
Can you remove the typos? Also properly format the Comments to make them look meaningful